packages/core/schematics/migrations/ngstyle-to-style-migration/util.ts

```diff
@@ -0,0 +1,457 @@
+): {
+  replacementCount: number;
+  migrated: string;
+  changed: boolean;
+} {
+  const parsed = parseTemplate(template);
+  if (!parsed.tree || !parsed.tree.rootNodes.length) {
+    return {migrated: template, changed: false, replacementCount: 0};
+  }
+
+  const visitor = new NgStyleCollector(template, componentNode, typeChecker);
+  visitAll(visitor, parsed.tree.rootNodes, config);
+
+  let newTemplate = template;
+  let changedOffset = 0;
+  let replacementCount = 0;
+
+  for (const {start, end, replacement} of visitor.replacements) {
+    const currentLength = newTemplate.length;
+
+    newTemplate = replaceTemplate(newTemplate, replacement, start, end, changedOffset);
+    changedOffset += newTemplate.length - currentLength;
+    replacementCount++;
+  }
+
+  return {migrated: newTemplate, changed: newTemplate !== template, replacementCount};
+}
+
+/**
+ * Creates a Replacement to remove `NgStyle` from a component's `imports` array.
+ * Uses ReflectionHost + PartialEvaluator for robust AST analysis.
+ */
+export function createNgStyleImportsArrayRemoval(
+  classNode: ts.ClassDeclaration,
+  file: ProjectFile,
+  typeChecker: ts.TypeChecker,
+): Replacement | null {
+  const reflector = new TypeScriptReflectionHost(typeChecker);
+  const evaluator = new PartialEvaluator(reflector, typeChecker, null);
+
+  // Use ReflectionHost to get decorators instead of manual AST traversal
+  const decorators = reflector.getDecoratorsOfDeclaration(classNode);
+  if (!decorators) {
+    return null;
+  }
+
+  // Find @Component decorator using ReflectionHost
+  const componentDecorator = decorators.find((decorator) => decorator.name === 'Component');
+
+  if (!componentDecorator || !componentDecorator.args || componentDecorator.args.length === 0) {
+    return null;
+  }
+
+  // Use PartialEvaluator to evaluate the decorator metadata
+  const decoratorMetadata = evaluator.evaluate(componentDecorator.args[0]);
+
+  if (!decoratorMetadata || typeof decoratorMetadata !== 'object') {
+    return null;
+  }
+
+  // Get the actual AST node for the imports property
+  const componentDecoratorNode = componentDecorator.node;
+  if (
+    !ts.isDecorator(componentDecoratorNode) ||
+    !ts.isCallExpression(componentDecoratorNode.expression) ||
+    !ts.isObjectLiteralExpression(componentDecoratorNode.expression.arguments[0])
+  ) {
+    return null;
+  }
+
+  const objLiteral = componentDecoratorNode.expression.arguments[0];
+  const importsProperty = objLiteral.properties.find(
+    (p): p is ts.PropertyAssignment =>
+      ts.isPropertyAssignment(p) &&
+      p.name.getText() === 'imports' &&
+      ts.isArrayLiteralExpression(p.initializer),
+  );
+
+  if (!importsProperty || !ts.isArrayLiteralExpression(importsProperty.initializer)) {
+    return null;
+  }
+
+  const importsArray = importsProperty.initializer;
+
+  const ngStyleIndex = importsArray.elements.findIndex(
+    (e): e is ts.Identifier => ts.isIdentifier(e) && e.text === ngStyleStr,
+  );
```

Comment:
```suggestion
  const ngStyleIndex = importsArray.elements.findIndex(
    (e) => ts.isIdentifier(e) && e.text === ngStyleStr,
  );
```
I don't think we need the type predicate here